### PR TITLE
Quantized ConvTranspose2d

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35460 Quantized ConvTranspose1d
* **#35371 Quantized ConvTranspose2d**
* #35370 Base class for the quantized ConvTranspose

Differential Revision: [D20641811](https://our.internmc.facebook.com/intern/diff/D20641811)

### CODE REVIEW

File: torch/nn/quantized/modules/conv.py [hunk 1]
```diff
@@ -563,3 +563,138 @@ def _output_padding(self, input, output_size, stride, padding, kernel_size):
...
+            cls._FLOAT_MODULE.__name__
+        assert hasattr(mod, 'qconfig'), \
+            'Input float module must have qconfig defined.'
+        weight_post_process = mod.qconfig.weight()
+        weight_post_process(mod.weight)
+        act_scale, act_zp = mod.activation_post_process.calculate_qparams()
+        assert weight_post_process.dtype == torch.qint8, \
+            'Weight observer must have a dtype of qint8'
+        qweight = _quantize_weight(mod.weight.float(), weight_post_process)
+        qconv = cls(mod.in_channels, mod.out_channels, mod.kernel_size,
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/conv.py]: I believe we have `output_padding` as well

File: torch/nn/quantized/modules/conv.py [hunk 2]
```diff
@@ -563,3 +563,157 @@ def _output_padding(self, input, output_size, stride, padding, kernel_size):
...
+
+    def _get_name(self):
+        return 'QuantizedConvTranpose2d'
+
+    def set_weight_bias(self, w, b):
+        # type: (torch.Tensor, Optional[torch.Tensor]) -> None
+
+        # TODO(z-a-f): Once the quantized `.flip` is implemented,
+        #              we can remove the requantization
+        qscale = w.q_scale()
```
Comment (jerryzh168) [inline torch/nn/quantized/modules/conv.py]: wait, what if weight is per channel quantized?

Comment (z-a-f) [reply torch/nn/quantized/modules/conv.py]: Not supported yet -- I will send a PR with the per-channel support later on

**Comment by z-a-f** [comment]
Coming up: unittests

**Pull request closed without merging**